### PR TITLE
[DRFT-290] Improve error handling for baseline systems api

### PR DESCRIPTION
**Pull request by jpramos123**
Now if users try to associate an already associated system to the baseline, our response code is `400` with a comprehensive message instead of `500`.

Request:
Assuming {{baseline_UUID}} is 
`POST: {{BASELINE_BASE_CI_URL}}/api/system-baseline/v1/baselines/{{baseline_UUID}}/systems`

Body:
`{
    "system_ids": [
  "{{SOME_SYSTEM_ID_ALREADY_ASSOCIATED}}"
    ]
}`

Response:
`Responde code: 400`
`{
    "message": "System(s) already associated with this baseline"
}`

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: system_baseline/models.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def to_json(self, withhold_facts=False, withhold_system_ids=True):
         return json_dict
 
     def add_mapped_system(self, system_id):
+        print("MAPPED SYSTEMS: ", self.mapped_systems)
```
Comment (fellipeh) [inline system_baseline/models.py:66]: Forgotten to remove this line?

File: system_baseline/views/v1.py [hunk 2]
```diff
@@ -661,6 +662,10 @@ def create_systems_with_baseline(baseline_id, body):
         message = str(error)
         current_app.logger.audit(message, request=request, success=False)
         raise HTTPError(HTTPStatus.BAD_REQUEST, message=message)
+    except IntegrityError:
+        message = "System(s) already associated with this baseline"
+        current_app.logger.audit(message, request=request, success=False)
```
Comment (fellipeh) [inline system_baseline/views/v1.py:667]: I believe it's better to put which system_id raised this error, since we are iterating over a `system_ids` list

**Review by fellipeh — CHANGES_REQUESTED** [review]
Thanks for your submissions. Some comments bellow.

File: system_baseline/models.py (hunk 1 above)
Comment (jpramos123) [reply system_baseline/models.py:66]: Forgot to remove :) 

Thanks for that

File: system_baseline/views/v1.py (hunk 2 above)
Comment (jpramos123) [reply system_baseline/views/v1.py:667]: Let me give it a try

Comment (sthirugn) [reply system_baseline/views/v1.py:667]: for my understanding - will this stop all systems from getting associated and quit this try block even if one system is already associated?

**Pull request closed without merging**